From 55ffd7d63657117ee7792c35675eb27b27325e89 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 30 Nov 2023 17:07:47 +0100
Subject: [PATCH 0368/1695] dts: stm32: add tamper configuration on stm32mp15x
 discovery boards

Add internal tampers configuration for stm32mp15x discovery boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5ea4659ebd4576c57be607c81a974e3293ba202a dts: stm32: add tamper configuration on stm32mp15x discovery boards]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: Ie943ff031b8a9ea85141e6e757886144c3769797
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/343137
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/349415
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 188194135..9c5a64acc 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -8,6 +8,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/mfd/st,stpmic1.h>
 #include <dt-bindings/soc/stm32mp15-etzpc.h>
+#include <dt-bindings/tamper/st,stm32-tamp.h>
 
 / {
 	memory@c0000000 {
@@ -691,6 +692,12 @@
 	status = "disabled";
 };
 
+&tamp {
+	status = "okay";
+	st,tamp-passive-nb-sample = <4>;
+	st,tamp-passive-sample-clk-div = <16384>;
+};
+
 &timers1 {
 	/* spare dmas for other usage */
 	/delete-property/dmas;
-- 
2.47.2

